### PR TITLE
Git styled -h and --issue options

### PR DESCRIPTION
**Pull request by lovef**
Use git-sh-setup (like git rebase) to get git styled help `-h` option.

Also added git styled `-i`, `--issue` options as aliases `issue` option

### CODE REVIEW

**Comment by lovef** [comment]
shellcheck didn't like sourcing git-sh-setup so I needed to disable that check to get a passing build, the option to add a constant path seems flaky to me as it may change in the build environment.

**Comment by paulirish** [comment]
first up, git-sh-setup is really useful. had no idea it existed!  
thanks for suggesting it.

image


----

one thing you're doing here is introducing the `git open -i` (or `--issue`) flag as an alternative to the `git open issue` command.  is that for brevity?  


Feels kind of odd to support **both** the flag and the command, so it feels better to decide on a winner and go with that. 

cc @moosichu @derimagia  who are `issue` users

**Comment by lovef** [comment]
Your welcome :)

I think the problem with the `issue` command is that it might be confused for a parameter. Consider that you might want to be able to support `git open [<remote>] [branch]`, making `remote` an optional parameter, then `issue` command might be confused for a branch parameter. `-i` and `--issue` are less ambiguous.

**Comment by derimagia** [comment]
Personally I agree with @lovef, Also that way if we wanted to support other things (like opening the commit for example, or maybe even a compare (something like) `git open -c v1.5.0...HEAD` or `git open -c v1.5.0` -> [v1.5.0...HEAD](https://github.com/paulirish/git-open/compare/v1.5.0...HEAD)

**Comment by derimagia** [comment]
Actually looks like this conversation also happened here: #64

**Comment by ffes** [comment]
Since we are working towards 2.0 with #100 breaking changes should not be a problem.

I prefer the `--issue` option as well. I think it is better then the `issue` command

It give you the possibility to do something like `git open upstream --issue`

And adding additional options like (just a thought) `--issues` to open the issues page of the repo could be very nice to have, just `--prs` or `--wiki` or `--tags`, although some of them are quite GitHub specific :wink:

**Comment by paulirish** [comment]
sg then! we'll make the change to --issue and kill off the `issue` command.

@lovef do you want to update this PR accordingly (and fix the conflict)?

**Comment by lovef** [comment]
Done, kept `-i` and `--issue` and removed `issue`

File: git-open [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
-if [[ "$1" == 'issue' ]]; then
-  is_issue=1
-
-  # Allow the user to provide other args, aka `git open issue upstream 79`
+while test $# != 0
+do
+  case "$1" in
+  --issue)  is_issue=1;;
+  --)
+    shift
```
Comment (paulirish) [inline git-open]: I believe you can condense this down a tad:

```sh
while test $# != 0; do
  case "$1" in
  --issue)  is_issue=1;;
  --) shift; break ;;
  esac
  shift
done
```

does that sound good to you?

**Comment by ffes** [comment]
Shouldn't there be a `--help` as well?

**Comment by ffes** [comment]
> Shouldn't there be a `--help` as well?

Never mind... Was AFK when I wrote that.

Now I see the difference between `-h` and `--help`.
`-h` shows a short help, `--help` opens the man page which we don't have.

**Comment by paulirish** [comment]
> -h shows a short help, 

actually it says "invalid option -h" and then prints the help.   this caught me out, too. :)

**Comment by lovef** [comment]
> actually it says "invalid option -h"

Does it? The screenshot you uploaded looked fine, and of course, it works on my machine ;)

**Comment by paulirish** [comment]
>> actually it says "invalid option -h"

> Does it? The screenshot you uploaded looked fine, and of course, it works on my machine ;)

oops! sorry. i got my wires crossed when remembering my testing yesterday. :)

**Comment by lovef** [comment]
Ok, is it ready to be merged then?

**Pull request merged**